### PR TITLE
Improving the writing testable code section

### PR DESCRIPTION
**Pull request by eduard13**
## Purpose of this pull request

This pull request (PR) improves the documentation of how to write testable code, by providing some simple of bad and good examples. Also it explains why it needs to keep it simple and maintainable.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.1/test/unit/writing_testable_code.html
- https://devdocs.magento.com/guides/v2.2/test/unit/writing_testable_code.html
- https://devdocs.magento.com/guides/v2.3/test/unit/writing_testable_code.html

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.1/test/unit/writing_testable_code.md [hunk 1]
```diff
@@ -151,6 +151,62 @@ Shorter methods do less, which in turn means they are easier to test. The same i
 
 As a rule of thumb, try to keep methods to five or fewer lines of code.
 
+### Each function has one function
```
Comment (rogyar) [inline guides/v2.1/test/unit/writing_testable_code.md]: I would recommend rephrasing it as `Each function has one single purpose` to avoid tautology :)

Comment (rogyar) [inline guides/v2.1/test/unit/writing_testable_code.md]: I would recommend rephrasing it as Each function has one single purpose to avoid tautology :)

Comment (eduard13) [reply guides/v2.1/test/unit/writing_testable_code.md]: @rogyar, agree with that.
Thank you.

**Comment by dobooth** [comment]
running tests

**Comment by dshevtsov** [comment]
@dobooth please add _whatnew_

**Pull request merged**